core/src/SYCL/Kokkos_SYCL_Parallel_Range.hpp

```diff
@@ -92,12 +103,24 @@ class Kokkos::Impl::ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>,
     sycl::queue& q = *instance.m_queue;
 
     auto parallel_for_event = q.submit([&](sycl::handler& cgh) {
-      FunctorWrapperRangePolicyParallelFor<Functor, Policy> f{policy.begin(),
-                                                              functor};
-      sycl::range<1> range(policy.end() - policy.begin());
       cgh.depends_on(memcpy_event);
-      cgh.parallel_for<FunctorWrapperRangePolicyParallelFor<Functor, Policy>>(
-          range, f);
+      if constexpr (Policy::launch_bounds::maxTperB == 0) {
+        FunctorWrapperRangePolicyParallelFor<Functor, Policy> f{policy.begin(),
+                                                                functor};
+        sycl::range<1> range(policy.end() - policy.begin());
+        cgh.parallel_for<FunctorWrapperRangePolicyParallelFor<Functor, Policy>>(
+            range, f);
+      } else {
+        const auto actual_range = policy.end() - policy.begin();
+        const auto wgroup_size  = Policy::launch_bounds::maxTperB;
+        const auto launch_range =
```

Comment:
global_size must be a multiple of local_size or else the kernel launch fails. Replace with 
```
        const auto launch_range =
            std::ceil(float(actual_range) / wgroup_size) * wgroup_size;
```

Reply:
How is this different from what I am doing already? I round `actual` to the next multiple of `wgroup_size`.